injectproxy/enforce.go

```diff
@@ -14,28 +14,33 @@
 package injectproxy
 
 import (
+	"errors"
 	"fmt"
 
 	"github.com/prometheus/prometheus/pkg/labels"
 	"github.com/prometheus/prometheus/promql/parser"
 )
 
 type Enforcer struct {
-	labelMatchers map[string]*labels.Matcher
+	labelMatchers  map[string]*labels.Matcher
+	errorOnReplace bool
 }
 
-func NewEnforcer(ms ...*labels.Matcher) *Enforcer {
+func NewEnforcer(errorOnReplace bool, ms ...*labels.Matcher) *Enforcer {
 	entries := make(map[string]*labels.Matcher)
 
 	for _, matcher := range ms {
 		entries[matcher.Name] = matcher
 	}
 
 	return &Enforcer{
-		labelMatchers: entries,
+		labelMatchers:  entries,
+		errorOnReplace: errorOnReplace,
 	}
 }
 
+var ErrIllegalLabelMatcher = errors.New("Label match value conflicts with injected value")
```

Comment:
In Golang we normally make errors start with a lower-case letter.
Also, for consistency this should be `matcher` I think 
```suggestion
var ErrIllegalLabelMatcher = errors.New("label matcher value conflicts with injected value")
```